tests/expressions_window/tests.py

```diff
@@ -0,0 +1,866 @@
+            ('Smith', 55000, 'Sales', datetime.date(2007, 6, 1), 53000),
+            ('Brown', 53000, 'Sales', datetime.date(2009, 9, 1), None),
+        ], transform=lambda row: (row.name, row.salary, row.department, row.hiredate, row.lead),)
+
+    @skipIf(connection.vendor == 'mysql', 'Lead not supported on MariaDB')
+    def test_lead_offset(self):
+        """
+        Determine what the person hired after someone makes. Due to
+        ambiguity, the name is also included in the ordering.
+        """
+        qs = WindowTestModel.objects.annotate(lead=Window(
+            expression=Lead('salary', offset=2),
+            partition_by='department',
+            order_by=[F('hiredate').asc(), F('name').asc()])
+        )
+
+        self.assertQuerysetEqual(qs, [
+            ('Jones', 45000, 'Accounting', datetime.date(2005, 11, 1), 37000),
+            ('Jenson', 45000, 'Accounting', datetime.date(2008, 4, 1), 50000),
+            ('Williams', 37000, 'Accounting', datetime.date(2009, 6, 1), None),
+            ('Adams', 50000, 'Accounting', datetime.date(2013, 7, 1), None),
+            ('Wilkinson', 60000, 'IT', datetime.date(2011, 3, 1), None),
+            ('Moore', 34000, 'IT', datetime.date(2013, 8, 1), None),
+            ('Johnson', 80000, 'Management', datetime.date(2005, 6, 1), None),
+            ('Miller', 100000, 'Management', datetime.date(2005, 6, 1), None),
+            ('Smith', 38000, 'Marketing', datetime.date(2009, 10, 1), None),
+            ('Johnson', 40000, 'Marketing', datetime.date(2012, 3, 1), None),
+            ('Smith', 55000, 'Sales', datetime.date(2007, 6, 1), None),
+            ('Brown', 53000, 'Sales', datetime.date(2009, 9, 1), None),
+        ], transform=lambda row: (row.name, row.salary, row.department, row.hiredate, row.lead),
+            ordered=False)
+
+    @skipIf(connection.vendor == 'mysql', 'Lead not supported on MariaDB')
+    def test_lead_default(self):
+        qs = WindowTestModel.objects.annotate(lead_default=Window(
+            expression=Lead(expression='salary', offset=5, default=60000),
+            partition_by=F('department'),
+            order_by=F('department').asc())
+        )
+
+        self.assertQuerysetEqual(qs, [
+            ('Williams', 37000, 'Accounting', 60000),
+            ('Jenson', 45000, 'Accounting', 60000),
+            ('Adams', 50000, 'Accounting', 60000),
+            ('Jones', 45000, 'Accounting', 60000),
+            ('Wilkinson', 60000, 'IT', 60000),
+            ('Moore', 34000, 'IT', 60000),
+            ('Johnson', 80000, 'Management', 60000),
+            ('Johnson', 40000, 'Marketing', 60000),
+            ('Miller', 100000, 'Management', 60000),
+            ('Smith', 38000, 'Marketing', 60000),
+            ('Brown', 53000, 'Sales', 60000),
+            ('Smith', 55000, 'Sales', 60000),
+        ], transform=lambda row: (row.name, row.salary, row.department, row.lead_default), ordered=False)
```

Comment:
I don't think we need to check all rows, probably sth like this:
```python
self.assertEqual(list(qs.values_list('lead_default', flat=True).distinct()), [60000])
```
will be sufficient. We have a similar situation in the `test_nth_returns_null`.